src/device.cpp

```diff
@@ -716,10 +722,36 @@ bool cvk_device::create_vulkan_queues_and_device(uint32_t num_queues,
     cvk_info("Creating Vulkan device and queues");
     // Give all queues the same priority
     std::vector<float> queuePriorities(num_queues, 1.0f);
+    void* pNext = nullptr;
+    VkDeviceQueueGlobalPriorityCreateInfoKHR globalPriorityCreateInfo;
+    cvk_info_fn("queue global priority: %u",
+                m_features_queue_global_priority.globalPriorityQuery);
+    if (m_features_queue_global_priority.globalPriorityQuery) {
```

Comment:
This does not guarantee that specific priorities are supported so queue creation might fail if a user passes a priority that the Vulkan implementation does not support. This might be fine in practice. If we want to do better we should use `VkQueueFamilyGlobalPriorityPropertiesKHR` to get the list of supported priorities and maybe fall back to one that's supported with a warning when a user requests an unsupported priority. Fine if you want to do this in this PR, fine if you don't.

Reply:
I have tried to use `VkQueueFamilyGlobalPriorityPropertiesKHR`, but all the drivers I have tested return no supported priority, while they do support at least medium. And sometimes even more.
I can add a warning, but I'm not fond of it as it would always be printed in ChromeOS while it would not mean anything.